File: challenges/14/solution.ts
function longestCommonPrefix(strs: string[]) {
    let x = strs[0];
    let output = ""

    for(let i = 0; i < x.length; i++) {
        for (let j = 1; j < strs.length; j++) {
            if(strs[j].length < i) return output;
            if(strs[j][i] !== x[i]) return output;
            else continue;
        }
        output += x[i];
    }
    return output;
};

File: challenges/14/solution.py
#https://leetcode.com/problems/longest-common-prefix/
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
 
        # sort strings
        strs = sorted(strs)

        # compare first string and last in order to find the common prefix
        first = strs[0]
        last = strs[-1]

        prefix = ""
        for i in range(len(first)):
    
            if(first[i] != last[i]):
                return prefix

            # common: add to prefix
            prefix += first[i]

        return prefix


